PhysicsPackages/IMCPhysicsPackage_class.f90

```diff
@@ -0,0 +1,507 @@
+
+  ! Physics package interface
+  use physicsPackage_inter,           only : physicsPackage
+
+  ! Geometry
+  use geometry_inter,                 only : geometry
+  use geometryReg_mod,                only : gr_geomPtr  => geomPtr, gr_addGeom => addGeom, &
+                                             gr_geomIdx  => geomIdx
+
+  ! Nuclear Data
+  use materialMenu_mod,               only : mm_nMat           => nMat ,&
+                                             mm_matName        => matName
+  use nuclearDataReg_mod,             only : ndReg_init        => init ,&
+                                             ndReg_activate    => activate ,&
+                                             ndReg_display     => display, &
+                                             ndReg_kill        => kill, &
+                                             ndReg_get         => get ,&
+                                             ndReg_getMatNames => getMatNames
+  use nuclearDatabase_inter,          only : nuclearDatabase
+  use IMCMaterial_inter,              only : IMCMaterial, IMCMaterial_CptrCast
+  use mgIMCMaterial_inter,            only : mgIMCMaterial
+
+  ! Operators
+  use collisionOperator_class,        only : collisionOperator
+  use transportOperator_inter,        only : transportOperator
+
+  ! Tallies
+  use tallyCodes
+  use tallyAdmin_class,               only : tallyAdmin
+  use tallyResult_class,              only : tallyResult
+  use imcWeightClerk_class,           only : imcWeightResult
+
+  ! Factories
+  use transportOperatorFactory_func,  only : new_transportOperator
+  use sourceFactory_func,             only : new_source
+
+  implicit none
+
+  private
+
+  !!
+  !! Physics Package for IMC calculations
+  !!
+  type, public,extends(physicsPackage) :: IMCPhysicsPackage
+    private
+    ! Building blocks
+    class(nuclearDatabase), pointer        :: nucData => null()
+    class(geometry), pointer               :: geom    => null()
+    integer(shortInt)                      :: geomIdx = 0
+    type(collisionOperator)                :: collOp
+    class(transportOperator), allocatable  :: transOp
+    class(RNG), pointer                    :: pRNG    => null()
+    type(tallyAdmin),pointer               :: tally   => null()
+    type(tallyAdmin),pointer               :: imcWeightAtch => null()
+
+    ! Settings
+    integer(shortInt)  :: N_cycles
+    integer(shortInt)  :: pop
+    integer(shortInt)  :: limit
+    real(defReal)      :: deltaT
+    character(pathLen) :: outputFile
+    character(nameLen) :: outputFormat
+    integer(shortInt)  :: printSource = 0
+    integer(shortInt)  :: particleType
+    logical(defBool)   :: sourceGiven = .false.
+    integer(shortInt)  :: nMat
+
+    ! Calculation components
+    type(particleDungeon), allocatable :: thisCycle
+    type(particleDungeon), allocatable :: nextCycle
+      ! Note that other physics packages used pointers for these particleDungeons ( => null() )
```

Comment:
Pointers are good because they should allow fast, easy flipping/swapping of empty and full dungeons.